ingest/validation/validate_metadata.py

```diff
@@ -1247,13 +1253,107 @@ def review_metadata_names(metadata):
             metadata.store_validation_issue("error", "metadata_name", error_msg)
 
 
+def identify_multiply_assigned(list):
+    """ Given a list of ontologyIDs and their purported labels,
+        return list of unique multiply-assigned labels
+    """
+    ontology_tracker = defaultdict(lambda: defaultdict(int))
+    multiply_assigned = []
+    for element in list:
+        id, label = element
+        ontology_tracker[label][id] += 1
+    for label in ontology_tracker:
+        if len(ontology_tracker[label].keys()) > 1:
+            multiply_assigned.append(label)
+    multiply_assigned.sort()
+    return multiply_assigned
+
+
+def assess_ontology_ids(ids, property_name, metadata):
+    """
+    Check ordered collection of ontology IDs for increasing numeric values
+    """
+    evidence_of_excel_drag = False
+    evidence_of_excel_drag_threshold = 25
+    binned_ids = defaultdict(list)
+    for id in ids:
+        try:
+            ontology_shortname, term_id = re.split("[_:]", id)
+            binned_ids[ontology_shortname].append(term_id)
+        except (ValueError, TypeError):
+            # invalid ontology id will already be flagged as convention error
+            # no need to also mark as ontology error as part of excel drag detection
+            pass
+    for ontology in binned_ids.keys():
+        id_numerics = []
+        incrementation_count = 0
+        for term in binned_ids[ontology]:
+            term_numeric = re.search('(\d)*$', term)
+            id_numerics.append(int(term_numeric.group()))
+        for x, y in zip(id_numerics, id_numerics[1:]):
```

Comment:
I think it would be great to have some comments on the logic here - I think I understand what's going on, but it's just obtuse enough that for someone like me with rather weak Python it's a little murky.

Reply:
added 64d11db